app/main.js

```diff
@@ -86,6 +125,37 @@ function launchDaemonIfNotRunning() {
   );
 }
 
+/*
+ * Last resort for killing unresponsive daemon instances.
+ * Looks for any processes called "lbrynet-daemon" and
+ * tries to force kill them.
+ */
+function forceKillAllDaemonsAndQuit() {
+  console.log('Attempting to force kill any running lbrynet-daemon instances...');
+
+  const fgrepOut = child_process.spawnSync('pgrep', ['-x', 'lbrynet-daemon'], {encoding: 'utf8'}).stdout;
+  const daemonPids = fgrepOut.match(/\d+/g);
+  if (!daemonPids) {
+    console.log('No lbrynet-daemon found running.');
+    quitNow();
```

Comment:
We could probably use Promises to simplify the shutdown process so we're not calling quitNow in so many places, but for now my goal is to make the logic extremely explicit and easy to reason about.